webapi/src/main/scala/org/knora/webapi/slice/common/repo/rdf/RdfModel.scala

```diff
@@ -0,0 +1,387 @@
+      chunk         <- getBooleanLiterals(propertyIri)
+      nonEmptyChunk <- ZIO.fromOption(NonEmptyChunk.fromChunk(chunk)).orElseFail(LiteralNotPresent(propertyIri))
+    } yield nonEmptyChunk
+
+  /**
+   * Returns the IRI of the object of a given predicate IRI.
+   *
+   * @param propertyIri the IRI of the predicate.
+   * @return            the [[InternalIri]] of the object or None if the object is not present.
+   */
+  def getObjectIri(propertyIri: String): IO[RdfError, Option[InternalIri]] =
+    getObjectUri(propertyIri).map(InternalIri).unsome
+
+  /**
+   * Returns the IRI of the object of a given predicate IRI.
+   * Fails if the object is not present.
+   *
+   * @param propertyIri the IRI of the predicate.
+   * @return            the [[InternalIri]] of the object or an [[RdfError]] if the object is not present.
+   */
+  def getObjectIriOrFail(propertyIri: String): IO[RdfError, InternalIri] =
+    getObjectIri(propertyIri).someOrFail(ObjectNotPresent(propertyIri))
+
+  /**
+   * Returns the IRIs of the objects of a given predicate IRI.
+   *
+   * @param propertyIri the IRI of the predicate.
+   * @return            the [[InternalIri]]s of the objects or an [[RdfError]] if the objects are not present.
+   */
+  def getObjectIris(propertyIri: String): IO[RdfError, Chunk[InternalIri]] =
+    getObjectUris(propertyIri).map(_.map(InternalIri))
+
+  /**
+   * Returns the IRIs of the objects of a given predicate IRI.
+   * Fails if the objects are not present.
+   *
+   * @param propertyIri the IRI of the predicate.
+   * @return            the [[InternalIri]]s of the objects or an [[RdfError]] if the objects are not present.
+   */
+  def getObjectIrisOrFail(propertyIri: String): IO[RdfError, NonEmptyChunk[InternalIri]] =
+    for {
+      chunk         <- getObjectIris(propertyIri)
+      nonEmptyChunk <- ZIO.fromOption(NonEmptyChunk.fromChunk(chunk)).orElseFail(ObjectNotPresent(propertyIri))
+    } yield nonEmptyChunk
+
+}
+
+/**
+ * Wrapper around Jena's [[Model]].
+ * Exposes access to resources of the model's graph.
+ */
+final case class RdfModel private (private val model: Model) {
+
+  /**
+   * Returns a [[RdfResource]] for the given subject IRI.
+   * Fails if no resource with the given IRI is present in the model.
+   *
+   * @param subjectIri the IRI of the resource.
+   * @return the [[RdfResource]] or an [[RdfError]] if the resource is not present.
+   */
+  def getResource(subjectIri: String): IO[RdfError, RdfResource] =
+    for {
+      resource <- ZIO.attempt(model.createResource(subjectIri)).orDie
+      _        <- ZIO.fail(ResourceNotPresent(subjectIri)).unless(resource.listProperties().hasNext)
+    } yield RdfResource(resource)
+
+}
+object RdfModel {
+
+  /**
+   * Creates an [[RdfModel]] from a turtle string.
+   * @param turtle the turtle string.
+   * @return the [[RdfModel]] or a throwable, if parsing the underlying Jena model failed.
+   */
+  def fromTurtle(turtle: String): Task[RdfModel] = ZIO.attempt {
```

Comment:
`StringReader` is a `Closeable`, I would recommend to manage that resource:

```
ZIO.scoped {
    val model = ModelFactory.createDefaultModel()
    for {
      reader <- ZIO.acquireRelease(ZIO.succeed(new StringReader(turtle)))(it => ZIO.succeed(it.close()))
      _      <- ZIO.attempt(model.read(reader, null, "TURTLE"))
    } yield RdfModel(model)
}
```
    
 Otherwise we might run into a small risk of memory leaks.

Reply:
good point, thanks